Add AWS Lambda client configuration options to documentation (#10345)
opensearch-project/documentation-website: _data-prepper/pipelines/configuration/processors/aws-lambda.md

`tags_on_match_failure` | List | Optional |  A list of tags to add to events when Lambda matching fails or encounters an unexpected error.
`sdk_timeout`        | Duration| Optional | Configures the SDK's client connection timeout period. Default is `60s`. 
`response_events_match` | Boolean | Optional | Specifies how Data Prepper interprets and processes Lambda function responses. Default is `false`.

>  `client`            | Object  | Optional | Client configuration

Reviewer: ```suggestion
 `client`            | Object  | Optional | The client configuration.
```

> `api_call_timeout`   | Duration | Optional | Api call timeout defines the time sdk maintains the api call to complete before timing out.

Reviewer: ```suggestion
`api_call_timeout`   | Duration | Optional | The time SDK maintains the API call to complete before timing out, in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Durations).
```

> `api_call_timeout`   | Duration | Optional | The time SDK maintains the API call to complete before timing out, in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Durations).

Reviewer 2: ```suggestion
`api_call_timeout`   | Duration | Optional | The amount of time that the SDK maintains the API call before timing out, in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Durations).
```

> `base_delay`         | Duration | Optional | Base delay for exponential backoff.

Reviewer: ```suggestion
`base_delay`         | Duration | Optional | The base delay for exponential backoff, in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Durations).
```

> `connection_timeout` | Duration | Optional | sdk timeout defines the time sdk maintains the connection to the client before timing out.

Reviewer: ```suggestion
`connection_timeout` | Duration | Optional | The time SDK maintains the connection to the client before timing out, in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Durations).
```

> `max_backoff`        | Duration | Optional | Maximum backoff time for exponential backoff. 

Reviewer: ```suggestion
`max_backoff`        | Duration | Optional | The maximum backoff time for exponential backoff, in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Durations).
```

> `connection_timeout` | Duration | Optional | The time SDK maintains the connection to the client before timing out, in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Durations).

Reviewer 2: ```suggestion
`connection_timeout` | Duration | Optional | The amount of time that the SDK maintains the connection to the client before timing out, in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Durations).
```

> `max_concurrency`    | Integer  | Optional | Max concurrency defined from the client side. 

Reviewer: ```suggestion
`max_concurrency`    | Integer  | Optional | The maximum concurrency defined on the client side. 
```

> `max_retries`        | Integer | Optional | Total retries we want before failing`.  

Reviewer: ```suggestion
`max_retries`        | Integer | Optional | The maximum number of retries before failing.  
```
